driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionProseTestSpecification.groovy

```diff
@@ -19,490 +19,159 @@ package com.mongodb.client
-      "subType": "00"
-    }
-  },
-  "creationDate": { "$date": { "$numberLong": "1232739599082000" } },
-  "updateDate": { "$date": { "$numberLong": "1232739599082000" } },
-  "status": { "$numberInt": "0" },
-  "masterKey": { "provider": "local" },
-  "keyAltNames": [ "altname1", "altname2" ]
-}
-''')
-
-    private final BsonDocument awsDataKeyDocument = BsonDocument.parse(
-            '''
-{
-    "_id": {
-        "$binary": {
-            "base64": "AAAAAAAAAAAAAAAAAAAAAA==",
-            "subType": "04"
-        }
-    },
-    "version": {
-        "$numberLong": "0"
-    },
-    "keyAltNames": [ "altname1", "altname2" ],
-    "keyMaterial": {
-        "$binary": {
-            "base64": "AQICAHhQNmWG2CzOm1dq3kWLM+iDUZhEqnhJwH9wZVpuZ94A8gG2Qei6UQdbOR5RWhPSrNwnAAAAwjCBvwYJKoZIhvcNAQcGoIGxMIGuAgEAMIGoBgkqhkiG9w0BBwEwHgYJYIZIAWUDBAEuMBEEDMJ2xcv8wKZzqTIX/gIBEIB7dNUvInthJHEd55QaEyVYSacoPvMlx2wzhxW+E6MBcfP+nCrzByLkqyHRhWs5NEvrOBT2nuc87iZIuK/WNR/pl5eK1xQ/8Cy0GrMfD3GIjYDlZ6aWc06cJvwvZd3Cgqx0pQnunwNr2EfStTGj7gHW23kzkfpxDiphqPnH",
-            "subType": "00"
-        }
-    },
-    "creationDate": {
-        "$date": {
-            "$numberLong": "1553026537755"
-        }
-    },
-    "updateDate": {
-        "$date": {
-            "$numberLong": "1553026537755"
-        }
-    },
-    "status": {
-        "$numberInt": "1"
-    },
-    "masterKey": {
-        "key": "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0",
-        "region": "us-east-1",
-        "provider": "aws"
-    }
-}'''
-    )
+    private final MongoCollection dataKeyCollection = getMongoClient()
+            .getDatabase(keyVaultNamespace.databaseName).getCollection(keyVaultNamespace.collectionName, BsonDocument)
+    private final MongoCollection<BsonDocument> dataCollection = getMongoClient()
+            .getDatabase(autoEncryptingCollectionNamespace.databaseName).getCollection(autoEncryptingCollectionNamespace.collectionName,
+            BsonDocument)
+    private final Map<String, Map<String, ? extends Object>> providerProperties =
+            ['local': ['key': Base64.getDecoder().decode('Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk')],
+             'aws'  : ['accessKeyId'    : System.getProperty('org.mongodb.test.awsAccessKeyId'),
+                       'secretAccessKey': System.getProperty('org.mongodb.test.awsSecretAccessKey')]
+            ]
 
     private MongoClient autoEncryptingClient
+    private ClientEncryption clientEncryption
+    private MongoCollection<BsonDocument> autoEncryptingDataCollection
 
     def setup() {
-        assumeFalse(isNotAtLeastJava8());
+        assumeFalse(isNotAtLeastJava8())
         assumeTrue(serverVersionAtLeast([4, 1, 10]))
```

Comment:
Technically, since we're not creating remote schemas I think we can get away without having this server version check. But as soon as we add a prose test configuring a remote schema we'll need this check. What do you think? Remove this line, or add a server version check to the prose test specification?

Reply:
I'll remove it for now.  And add it back if we change the prose test requirements.  Would a server version check add useful coverage to any of the client-side components tested here?  I'm not sure if it would.

Reply:
Removed server version check